### PR TITLE
r/iam_role: support for tags

### PR DESCRIPTION
**Pull request by kl4w**
References #6492
Requires: #6497

Changes proposed in this pull request:

* support for tags

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSIAMRole_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSIAMRole_ -timeout 120m
=== RUN   TestAccAWSIAMRole_importBasic
=== PAUSE TestAccAWSIAMRole_importBasic
=== RUN   TestAccAWSIAMRole_basic
=== PAUSE TestAccAWSIAMRole_basic
=== RUN   TestAccAWSIAMRole_basicWithDescription
=== PAUSE TestAccAWSIAMRole_basicWithDescription
=== RUN   TestAccAWSIAMRole_namePrefix
=== PAUSE TestAccAWSIAMRole_namePrefix
=== RUN   TestAccAWSIAMRole_testNameChange
=== PAUSE TestAccAWSIAMRole_testNameChange
=== RUN   TestAccAWSIAMRole_badJSON
=== PAUSE TestAccAWSIAMRole_badJSON
=== RUN   TestAccAWSIAMRole_disappears
=== PAUSE TestAccAWSIAMRole_disappears
=== RUN   TestAccAWSIAMRole_force_detach_policies
=== PAUSE TestAccAWSIAMRole_force_detach_policies
=== RUN   TestAccAWSIAMRole_MaxSessionDuration
=== PAUSE TestAccAWSIAMRole_MaxSessionDuration
=== RUN   TestAccAWSIAMRole_PermissionsBoundary
=== PAUSE TestAccAWSIAMRole_PermissionsBoundary
=== RUN   TestAccAWSIAMRole_tags
=== PAUSE TestAccAWSIAMRole_tags
=== CONT  TestAccAWSIAMRole_importBasic
=== CONT  TestAccAWSIAMRole_disappears
=== CONT  TestAccAWSIAMRole_force_detach_policies
=== CONT  TestAccAWSIAMRole_PermissionsBoundary
=== CONT  TestAccAWSIAMRole_tags
=== CONT  TestAccAWSIAMRole_basicWithDescription
=== CONT  TestAccAWSIAMRole_MaxSessionDuration
=== CONT  TestAccAWSIAMRole_namePrefix
=== CONT  TestAccAWSIAMRole_badJSON
=== CONT  TestAccAWSIAMRole_testNameChange
=== CONT  TestAccAWSIAMRole_basic
--- PASS: TestAccAWSIAMRole_badJSON (1.52s)
--- PASS: TestAccAWSIAMRole_disappears (6.62s)
--- PASS: TestAccAWSIAMRole_basic (8.32s)
--- PASS: TestAccAWSIAMRole_namePrefix (8.33s)
--- PASS: TestAccAWSIAMRole_importBasic (9.08s)
--- PASS: TestAccAWSIAMRole_force_detach_policies (9.51s)
--- PASS: TestAccAWSIAMRole_tags (13.31s)
--- PASS: TestAccAWSIAMRole_MaxSessionDuration (14.03s)
--- PASS: TestAccAWSIAMRole_testNameChange (15.94s)
--- PASS: TestAccAWSIAMRole_basicWithDescription (18.29s)
--- PASS: TestAccAWSIAMRole_PermissionsBoundary (30.97s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	31.015s
```

### CODE REVIEW

**Comment by bflad** [comment]
Once the changes from #6497 have been ported over to this pull request, will take a look. Thanks!

**Comment by kl4w** [comment]
@bflad rebased on top of extra changes in #6497 and extra commit for same reasons as per other review

```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSIAMRole_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSIAMRole_ -timeout 120m
=== RUN   TestAccAWSIAMRole_importBasic
=== PAUSE TestAccAWSIAMRole_importBasic
=== RUN   TestAccAWSIAMRole_basic
=== PAUSE TestAccAWSIAMRole_basic
=== RUN   TestAccAWSIAMRole_basicWithDescription
=== PAUSE TestAccAWSIAMRole_basicWithDescription
=== RUN   TestAccAWSIAMRole_namePrefix
=== PAUSE TestAccAWSIAMRole_namePrefix
=== RUN   TestAccAWSIAMRole_testNameChange
=== PAUSE TestAccAWSIAMRole_testNameChange
=== RUN   TestAccAWSIAMRole_badJSON
=== PAUSE TestAccAWSIAMRole_badJSON
=== RUN   TestAccAWSIAMRole_disappears
=== PAUSE TestAccAWSIAMRole_disappears
=== RUN   TestAccAWSIAMRole_force_detach_policies
=== PAUSE TestAccAWSIAMRole_force_detach_policies
=== RUN   TestAccAWSIAMRole_MaxSessionDuration
=== PAUSE TestAccAWSIAMRole_MaxSessionDuration
=== RUN   TestAccAWSIAMRole_PermissionsBoundary
=== PAUSE TestAccAWSIAMRole_PermissionsBoundary
=== RUN   TestAccAWSIAMRole_tags
=== PAUSE TestAccAWSIAMRole_tags
=== CONT  TestAccAWSIAMRole_importBasic
=== CONT  TestAccAWSIAMRole_disappears
=== CONT  TestAccAWSIAMRole_badJSON
=== CONT  TestAccAWSIAMRole_PermissionsBoundary
=== CONT  TestAccAWSIAMRole_tags
=== CONT  TestAccAWSIAMRole_namePrefix
=== CONT  TestAccAWSIAMRole_testNameChange
=== CONT  TestAccAWSIAMRole_MaxSessionDuration
=== CONT  TestAccAWSIAMRole_force_detach_policies
=== CONT  TestAccAWSIAMRole_basic
=== CONT  TestAccAWSIAMRole_basicWithDescription
--- PASS: TestAccAWSIAMRole_badJSON (1.54s)
--- PASS: TestAccAWSIAMRole_disappears (6.76s)
--- PASS: TestAccAWSIAMRole_namePrefix (8.56s)
--- PASS: TestAccAWSIAMRole_basic (8.58s)
--- PASS: TestAccAWSIAMRole_importBasic (9.25s)
--- PASS: TestAccAWSIAMRole_force_detach_policies (9.71s)
--- PASS: TestAccAWSIAMRole_tags (13.51s)
--- PASS: TestAccAWSIAMRole_MaxSessionDuration (14.17s)
--- PASS: TestAccAWSIAMRole_testNameChange (16.32s)
--- PASS: TestAccAWSIAMRole_basicWithDescription (18.19s)
--- PASS: TestAccAWSIAMRole_PermissionsBoundary (30.59s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	30.636s
```

File: website/docs/r/iam_role.html.markdown [hunk 1]
```diff
@@ -31,6 +31,10 @@ resource "aws_iam_role" "test_role" {
   ]
 }
 EOF
+
+  tags {
```
Comment (bflad) [inline website/docs/r/iam_role.html.markdown]: Nit: Missing equals sign, I'll fix on merge. 👍

**Comment by bflad** [comment]
This has been released in [version 1.46.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1460-november-20-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**